packages/headless/src/features/search/search-actions.ts

```diff
@@ -169,7 +167,7 @@ export const executeSearch = createAsyncThunk<
         queryCorrections: fetched.response.success.queryCorrections,
       },
       automaticallyCorrected: true,
-      analyticsAction: logDidYouMeanAutomatic(),
+      analyticsActions: [analyticsAction, logDidYouMeanAutomatic()],
```

Comment:
This seems like that only place in the PR where `analyticsActions` is an array with more than one action, so I am wondering if the switch to an array is the right approach.

The jsdoc for the `analyticsAction` (now called `analyticsActions`) property says:
`/** The analytics action to log after the query. */`

Couldn't we do this on line 160 though after the `automaticallyRetryQueryWithCorrection` promise has completed and we have made sure it was not an error?

What is the benefit of returning the action in the payload and to have it be processed by the middleware?

Reply:
Yeah, this is why I was hesitant too. Unfortunately, I don't see a shorter solution and equally efficient solution.